curated_transformers/generation/logits.py

```diff
@@ -121,3 +122,43 @@ def _process_logits(self, logits: Tensor):
             return
 
         logits /= self.temperature
+
+
+class MaskTransform(LogitsTransform):
+    """
+    Set the probability of specific classes to zero.
+    """
+
+    def __init__(self, classes_to_mask: List[int]):
+        """
+        Construct a mask logits transform.
+
+        :param classes_to_mask:
+            Identifers pertaining to the classes that need to be masked.
+            An empty list results in a no-op.
+        """
+        super().__init__()
+        self.classes_to_mask = torch.tensor(
+            classes_to_mask, dtype=torch.long
+        )  # `torch.int32` isn't supported in indexing operations prior in torch<2.0.0.
+        if self.classes_to_mask.dim() != 1:
+            raise ValueError("Class selector tensor for mask transform must be 1D")
+        elif (
+            self.classes_to_mask.size(dim=-1) != 0
+            and int(self.classes_to_mask.min(dim=-1).values) < 0
+        ):
+            raise ValueError("Class indentifiers for mask transform must be >= 0")
+
+    def _process_logits(self, logits: Tensor):
+        if self.classes_to_mask.size(dim=-1) == 0:
+            return
+
+        max_expected = logits.size(dim=-1)
+        max_received = int(self.classes_to_mask.max(dim=-1).values)
+        if max_received >= max_expected:
+            raise ValueError(
+                f"Class identifiers for mask transform must be < {max_expected}, but got {max_received}"
+            )
```

Comment:
I think Torch reports out of bound indexing. Good error messages are nice, but this does launch another kernel. I think it's worth looking wether the indexing below can be wrapped in a `try...except`, so that we don't have to go over the array twice.

Reply:
Switched to `try...except` .